### PR TITLE
Campt Error App tests using TestCube fixture

### PR DESCRIPTION
**Pull request by paarongiroux**
## Description

Added two gtests to test campt errors. Created new campt function to take in a pointer to a cube.

## Related Issue

gtest conversion

## How Has This Been Tested?

new gtests and existing app tests pass

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/CamptTests.cpp
Comment (Kelvinrr) [inline isis/tests/CamptTests.cpp]: Should probably change back to just `options`

File: isis/tests/CamptTests.cpp
Comment (Kelvinrr) [inline isis/tests/CamptTests.cpp]: Same here

**Review by Kelvinrr — CHANGES_REQUESTED** [review]

**Review by Kelvinrr — DISMISSED** [review]

**Comment by Kelvinrr** [comment]
Just going to wait for CI to finish.

**Comment by Kelvinrr** [comment]
A new test seems to be failing on CI so I'll leave this open for now.

**Comment by Kelvinrr** [comment]
Things are passing, Kaitlyn added a new test case, and this should be good to go as is. This has changes to the defaultcube fixture that others may need so we should merge after approval.  As other test cases get finished they should be PR-able.

File: isis/tests/CamptFunctionalTests.cpp
Comment (acpaquette) [inline isis/tests/CamptFunctionalTests.cpp]: I would change this back to APP_XML and update with the static keyword.

File: isis/tests/data/defaultImage/defaultCube.isd
Comment (acpaquette) [inline isis/tests/data/defaultImage/defaultCube.isd:1]: This concerns me but I understand why we need to have a real ISD for anything that needs a camera.

**Review by acpaquette — DISMISSED** [review]
Some comments and one change but everything looks pretty good :+1:

File: isis/tests/CamptFunctionalTests.cpp
Comment (paarongiroux) [inline isis/tests/CamptFunctionalTests.cpp]: A few (5) of these `EXPECT_DOUBLE_EQ` lines are failing when I run them locally. The differences seem insignificant, for example: 
```
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("SpacecraftAzimuth")
1679:     Which is: 240.08514246636
1679:   240.08514246657
1679:     Which is: 240.08514246657001
```

Maybe we should use EXPECT_NEAR instead?

Comment (paarongiroux) [reply isis/tests/CamptFunctionalTests.cpp]: Full list of failed EXPECT_DOUBLE_EQ's:
```
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("SpacecraftAzimuth")
1679:     Which is: 240.08514246636
1679:   240.08514246657
1679:     Which is: 240.08514246657001
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:113: Failure
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("SubSolarAzimuth")
1679:     Which is: 92.033828156504001
1679:   92.033828156965
1679:     Which is: 92.033828156965001
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:121: Failure
1679: Expected equality of these values:
1679:   (double) groundPoint.findKeyword("NorthAzimuth")
1679:     Which is: 332.65918494109002
1679:   332.65918493997
1679:     Which is: 332.65918493996998
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:136: Failure
1679: Expected equality of these values:
1679:   toDouble(groundPoint.findKeyword("LookDirectionCamera")[0])
1679:     Which is: -0.000127447324380692
1679:   -1.27447324380581e-04
1679:     Which is: -0.000127447324380581
1679: /home/pgiroux/repos/ISIS3/isis/tests/CamptFunctionalTests.cpp:137: Failure
1679: Expected equality of these values:
1679:   toDouble(groundPoint.findKeyword("LookDirectionCamera")[1])
1679:     Which is: 2.5816511718734801e-05
1679:   2.5816511718707e-05
1679:     Which is: 2.5816511718707001e-05
```

File: isis/tests/CamptFunctionalTests.cpp
Comment (Kelvinrr) [inline isis/tests/CamptFunctionalTests.cpp:75]: Any reason to change to expect_near? Was the precision to low?

Comment (kaitlyndlee) [reply isis/tests/CamptFunctionalTests.cpp:75]: Yes, it looks like there are just minor differences between mac and linux.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**
